File: PitRadio.Api/Data/Model/Song.cs
using System;

namespace PitRadio.Api.Data.Model
{
    public class Song
    {
        public Song(string uuid, string title, string file)
        {
            UUID = uuid;
            Title = title;
            File = file;
        }

        public string UUID { get; }
        public string Title { get; }
        public string File { get; }
    }
}

File: frontend/lib/song.dart
import 'package:json_annotation/json_annotation.dart';

part 'song.g.dart';

@JsonSerializable()
class Song {
  Song({required this.title, required this.file, this.uuid});

  final String? uuid;
  final String title;
  final String file;
  // final String author;
  // final String thumbnailUrl;

  factory Song.fromJson(Map<String, dynamic> json) => _$SongFromJson(json);
  Map<String, dynamic> toJson() => _$SongToJson(this);
}

typedef SongCallback = void Function(Song);
